pkg/operator/oidcdiscoveryendpoint/controller.go

```diff
@@ -0,0 +1,413 @@
+	if err != nil {
+		return err
+	}
+
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isServiceAccountTokenSigner(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isServiceAccountTokenSigner(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			// Deleting the token signer configmap shouldn't be done.
+			// Be safe and don't react.  When the openshift-kube-apiserver-operator
+			// recreates it, we'll catch it.
+			return false
+		},
+	}
+
+	// Create a namespace local cache separate from the Manager cache
+	// A namespace scoped cache can still handle cluster scoped resources
+	cache, err := cache.New(config, cache.Options{Namespace: tokenSignerConfigMapNamespace})
+	if err != nil {
+		return err
+	}
+
+	informer, err := cache.GetInformer(context.TODO(), &corev1.ConfigMap{})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+	if err != nil {
+		return err
+	}
+
+	mgr.Add(&oidcDiscoveryEndpointController{reconciler: r, cache: cache, logger: logger})
+
+	return nil
+}
+
+// openshift-kube-apiserver/bound-sa-token-signing-certs[service-account-xxx.pub]
+func isServiceAccountTokenSigner(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == tokenSignerConfigMapName {
+		return true
+	}
+	return false
+}
+
+type s3EndpointReconciler struct {
+	kubeclientset      *kubernetes.Clientset
+	configclientset    *configset.Clientset
+	logger             log.FieldLogger
+	eventRecorder      events.Recorder
+	infrastructureName string
+	region             string
+}
+
+var _ reconcile.Reconciler = &s3EndpointReconciler{}
+
+func (r *s3EndpointReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("reconciling AWS S3 OIDC discovery endpoint on detected change to service account token signer")
+
+	skip, err := r.reconcileClusterAuthentication()
+	if err != nil {
+		r.logger.Error(err)
+		return reconcile.Result{Requeue: true}, err
+	}
+	if skip {
+		return reconcile.Result{}, nil
+	}
+
+	err = r.reconcileS3Endpoint()
+	if err != nil {
+		r.logger.Error(err)
+		return reconcile.Result{Requeue: true}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *s3EndpointReconciler) reconcileClusterAuthentication() (bool, error) {
+	r.logger.Debugf("reconciling cluster Authentication serviceAccountIssuer")
+
+	auth, err := r.configclientset.ConfigV1().Authentications().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	serviceAccountIssuer := r.getIssuerURL()
+	if auth.Spec.ServiceAccountIssuer != "" {
+		r.logger.Debug("Authentication ServiceAccountIssuer already specified")
+		if auth.Spec.ServiceAccountIssuer != serviceAccountIssuer {
+			r.logger.Debug("Authentication ServiceAccountIssuer is not set to the s3 bucket location, skipping OIDC reconciliation")
```

Comment:
Should this be info or warn instead of debug?

Reply:
it is a persistent condition if the customer sets their own `ServiceAccountIssuer` so I didn't want it to be a warning since it just indicates an expected condition.  It would print each time the bound service account signing key rotated (I'm not sure how often that is off the top of my head).

i could make it an `info` if you like?